src/recon.rs

```diff
@@ -1,4 +1,5 @@
 #![deny(unsafe_op_in_unsafe_fn)]
+#![allow(non_camel_case_types)]
```

Comment:
We already have this in `lib.rs`; it's not needed here.  If you're seeing `rust-analyzer` warnings, that's a `rust-analyzer` issue.  It got fixed last time we upgraded our `nughtly` but then it came back at some point, so I'm not sure what the underlying issue is.